sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationHeadMockTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Azure.Core.Pipeline;
+using Azure.Core.Testing;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Http;
+using NUnit.Framework;
+using System;
+
+namespace Azure.Data.AppConfiguration.Tests
+{
+    [TestFixture(true)]
+    [TestFixture(false)]
+    public partial class ConfigurationMockTests: ClientTestBase
+    {
+        class HeadRequestPolicy : SynchronousHttpPipelinePolicy
+        {
+            public readonly static HttpPipelinePolicy Shared = new HeadRequestPolicy();
+
+            private HeadRequestPolicy() { }
+
+            public override void OnSendingRequest(HttpPipelineMessage message)
+                => message.Request.Method = RequestMethod.Head;          
+        }
+
+        class HeadRequestTransport : MockTransport
+        {
+            static readonly Func<MockRequest, MockResponse> Impl = (MockRequest) =>
+            {
+                var response = new MockResponse(200);
+                response.ContentStream = Stream.Null;
+                return response;
+            };
+
+            public readonly static HttpPipelineTransport Shared = new HeadRequestTransport();
+
+            private HeadRequestTransport() : base(Impl) {
+            }
+        }
+
+        [Test]
+        public async Task HeadRequest()
+        {
+            ConfigurationClient service = CreateTestService(HeadRequestTransport.Shared, headRequests: true);
+
+            Response<ConfigurationSetting> typedRetting = await service.GetAsync(s_testSetting.Key);
```

Comment:
What's the scenario for this -- you want to check whether a setting exists without getting the setting?  And you want to do that globally for every operation since we're designing this as a policy?

In Storage we expose this as two separate APIs where `HEAD` maps to `blob.GetPropertiesAsync(...)` and `GET` maps to `blob.DownloadAsync(...)`.

Reply:
In app config client, issuing HEAD requests seems to be very advanced scenarios, so I would not like to introduce new APIs for it.

Also, there is no "properties" model type, and the approach from storage would kind of force us to create such type.

Reply:
I feel like AppConfig isn't exactly burdening its users with an overabundance of scenarios.  It seems kind of weird to hide this scenario behind a pipeline policy that nulls out significant parts of the service's only model type.

But even if that's right for AppConfig, I'm not sure there are a lot of other services where we'd want this behavior.  I think in most cases it would be better to add additional APIs rather than have policies dramatically alter the semantics of existing APIs.

Reply:
I think we want additional/specialized APIs if issuing HEAD requests is a mainline scenario. If it's not, I think it's just not worth polluting the OM with these additional methods.

Reply:
Then put it in a `.Specialized` namespace?  Or if it's really so niche only the service team needs to use it, let them write their own `HeadRequestPolicy`?

Reply:
+1 on additional, dedicated APIs for HEAD requests if and where it makes sense.

Reply:
> Then put it in a .Specialized namespace? Or if it's really so niche only the service team needs to use it, let them write their own HeadRequestPolicy?

The namespace with policies is already very "specialized". Also, it's not enough to write the policy. We would also need to make sure that our serializers don't fail on empty content.